### PR TITLE
Fisrt implementation

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: src/MessageBodyRenderer.php [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+namespace Yiisoft\Mailer\View;
+
+use Yiisoft\Mailer\MessageInterface;
+use Yiisoft\View\Exception\ViewNotFoundException;
+use Yiisoft\View\View;
+
+/**
+ * View renderer used to compose message body.
+ *
+ * @api
```
Comment (samdark) [inline src/MessageBodyRenderer.php:14]: What is it for?

**Review by samdark — APPROVED** [review]

File: src/MessageBodyRenderer.php [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+     * @return string The text body.
+     */
+    private function generateTextBodyFromHtml(string $html): string
+    {
+        if (preg_match('~<body[^>]*>(.*?)</body>~is', $html, $match)) {
+            $html = $match[1];
+        }
+
+        // remove style and script
+        $html = preg_replace('~<((style|script))[^>]*>(.*?)</\1>~is', '', $html);
```
Comment (Tigrov) [inline src/MessageBodyRenderer.php:178]:
```suggestion
        $html = preg_replace('~<(style|script)[^>]*>.*?</\1>~is', '', $html);
```

File: src/MessageBodyRenderer.php [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+        // strip all HTML tags and decode HTML entities
+        $text = html_entity_decode(strip_tags($html), ENT_QUOTES | ENT_HTML5);
+
+        // improve whitespace
+        $text = preg_replace("~^[ \t]+~m", '', trim($text));
```
Comment (Tigrov) [inline src/MessageBodyRenderer.php:184]: Does it make sense to remove spaces from the end of lines?

```suggestion
        $text = preg_replace("~^[ \t]+|[ \t]+$~m", '', trim($text));
```

**Review by Tigrov — APPROVED** [review]

File: composer.json [hunk 4]
```diff
@@ -29,7 +29,9 @@
     "minimum-stability": "dev",
     "prefer-stable": true,
     "require": {
-        "php": "^8.1"
+        "php": "^8.1",
+        "yiisoft/mailer": "dev-extract-compose",
```
Comment (vjik) [inline composer.json]:
```suggestion
        "yiisoft/mailer": "dev-master",
```

**Pull request merged**